main.js

```diff
@@ -2,5 +2,127 @@
  * Created by Jenya on 18.04.15.
  */
 
+/// At first I'm creating general BeingClass. It will be the top of all being classes.
+function BeingClass(clas) {
+    this.clas = clas;
+    this.movementType = move.run;
+    this.legs = 0
+}
+
+// Than I'm adding move function to all beings.
+BeingClass.prototype.makeMove = function () {
+    this.movementType.move(this.name);
+    return this;
+};
+
+BeingClass.prototype.setLegsAmount = function (legsAmount) {
+    this.legs = legsAmount;
+    return this;
+};
+
+//Making a method to print object's properties
+BeingClass.prototype.logName = function () {
+    console.log("This.class: " + this.clas + "; this.name: " + this.name + "; this.legs= " + this.legs);
+    return this;
+};
+
+//Now I'm setting movement methods, strategy and type switch - BEGIN
+BeingClass.prototype.setMovementType = function (newMovementType) {
+    this.movementType = newMovementType;
+    return this;
+};
+
+var Movement = function (func) {
```

Comment:
as i said before it looks too complicated. i would simple use movement as a function